x-pack/plugins/security_solution/public/management/hooks/artifacts/use_get_updated_tags.test.tsx

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useGetUpdatedTags } from '.';
+import { renderHook } from '@testing-library/react-hooks';
+import type { ExceptionListItemSchema } from '@kbn/securitysolution-io-ts-list-types';
+import type { TagFilter } from '../../../../common/endpoint/service/artifacts/utils';
+
+describe('useGetUpdatedTags hook', () => {
+  const firstFilter: TagFilter = (tag) => tag.startsWith('first:');
+  const secondFilter: TagFilter = (tag) => tag.startsWith('second:') || tag === 'special_second';
+  const thirdFilter: TagFilter = (tag) => tag.startsWith('third:');
+
+  const filtersInOrder = {
+    first: firstFilter,
+    second: secondFilter,
+    third: thirdFilter,
+  };
+
+  const exampleException: Partial<ExceptionListItemSchema> = {
+    tags: [
+      'first:mozzarella',
+      'first:roquefort',
+
+      'second:cabernet',
+      'second:shiraz',
+      'special_second',
+
+      'third:tagliatelle',
+      'third:penne',
+    ],
+  };
```

Comment:
Suggestion: implement these as getter functions instead of mutable constant data. It will avoid having t he data changed (unexpectedly) from one test to the other.

Reply:
that's a good point, thanks! i guess i took it granted that the input is not mutated, but... well... nothing's granted in sw :D
b5162b2634bea9c343d42e9fb2a084f20b10b6c1
a21bec31b032a6cd796e8ac030fcd422f42cc9e8